runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -699,13 +692,6 @@ handleServerMessage(JITServer::ClientStream *client, TR_J9VM *fe, JITServer::Mes
          break;
       case MessageType::VM_getArrayLengthInElements:
          {
-         if (compInfoPT->getLastLocalGCCounter() != compInfoPT->getCompilationInfo()->getLocalGCCounter())
-            {
-            // GC happened, fail compilation
-            auto comp = compInfoPT->getCompilation();
-            comp->failCompilation<TR::CompilationInterrupted>("Compilation interrupted due to GC");
-            }
-
          uintptrj_t objectPointer = std::get<0>(client->getRecvData<uintptrj_t>());
          client->write(response, fe->getArrayLengthInElements(objectPointer));
```

Comment:
`fe->getArrayLengthInElements` assumes VM access is already held, but I don't think the client will have it here, so we need to acquire is briefly